### PR TITLE
[monorepo] fix: remove shell scripts

### PR DESCRIPTION
**Pull request by Wayonb**
Remove the shell scripts for docker and npm publishing.

### CODE REVIEW

File: jenkins/shared-library/vars/publish.groovy [hunk 1]
```diff
@@ -10,44 +10,53 @@ String getResource(String fileName) {
 	return libraryResource(fileName)
 }
 
-void copyFileListToTemp(String[] files) {
-	files.each { filepath ->
-		fileHelper.copyToTempFile(getResource(Paths.get('artifacts').resolve(filepath).toString()), filepath)
-	}
+String readNpmPackageNameVersion() {
+	Object packageJson = readJSON file: 'package.json'
+    return "${packageJson.name}@${packageJson.version}"
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/publish.groovy:15]: spaces => tabs

File: jenkins/shared-library/vars/publish.groovy [hunk 2]
```diff
@@ -10,44 +10,53 @@ String getResource(String fileName) {
...
+	if (config.publisher != 'npm') {
+		return
+	}
+
+	String npmPublishCommand = 'npm publish'
+	if (isAlphaRelease(phase)) {
+		npmPublishCommand += ' --tag alpha'
+	}
+
+	fileHelper.copyToLocalFile(getResource('artifacts/configuration/npmrc'), "${HOME}/.npmrc")
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/publish.groovy:56]: do we need this in artifacts of should we just create it on the fly?

File: jenkins/shared-library/vars/publish.groovy [hunk 3]
```diff
@@ -57,7 +66,7 @@ void pythonPublisher(Map config, String phase) {
 	}
 
 	credentialsId = PYTHON_CREDENTIALS_ID
-	if (phase == 'alpha') {
+	if (isAlphaRelease(phase)) {
 		credentialsId = TEST_PYTHON_CREDENTIALS_ID
 	}
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/publish.groovy:71]: any reason not to replace pypt script too?

File: jenkins/shared-library/vars/publish.groovy [hunk 4]
```diff
@@ -10,44 +10,53 @@ String getResource(String fileName) {
...
+	return readFile(file: 'version.txt').trim()
 }
 
-void copyWithJenkinsFunctionFileToTemp(String[] files) {
-	copyFileListToTemp(files + ['scripts/jenkins-functions.sh'] as String[])
+Boolean isAlphaRelease(String phase) {
+	return phase == 'alpha'
 }
 
 void dockerPublisher(Map config, String phase) {
```
[the remark this answers is not in the document]
Comment (Wayonb) [reply jenkins/shared-library/vars/publish.groovy:26]: Also explorer, faucet statistic service, and desktop wallet currently.

File: jenkins/shared-library/vars/publish.groovy (hunk 2 above)
Comment (Wayonb) [reply jenkins/shared-library/vars/publish.groovy:56]: I can look into creating it on the fly

File: jenkins/shared-library/vars/publish.groovy (hunk 3 above)
Comment (Wayonb) [reply jenkins/shared-library/vars/publish.groovy:71]: Nope, can remove the pypi script also

File: jenkins/shared-library/vars/publish.groovy [hunk 5]
```diff
@@ -6,48 +6,58 @@ void call(Map config, String phase) {
...
+		return
+	}
+
+	String npmPublishCommand = 'npm publish'
+	if (isAlphaRelease(phase)) {
+		npmPublishCommand += ' --tag alpha'
+	}
+
+	writeFile(file:".npmrc", text:  '//registry.npmjs.org/:_authToken=${NPM_TOKEN}')
+	runScript('cat .npmrc')
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/publish.groovy:57]: does this show up in jenkins log?

File: jenkins/shared-library/vars/publish.groovy [hunk 6]
```diff
@@ -57,13 +67,27 @@ void pythonPublisher(Map config, String phase) {
 	}
 
 	credentialsId = PYTHON_CREDENTIALS_ID
-	if (phase == 'alpha') {
+	if (isAlphaRelease(phase)) {
 		credentialsId = TEST_PYTHON_CREDENTIALS_ID
 	}
 
-	String scriptFullFilepath = fileHelper.copyToTempFile(getResource('artifacts/scripts/pypi-publish.sh'), 'scripts/pypi-publish.sh')
+	Object file = readFile 'requirements.txt'
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/publish.groovy:74]: file => requirementsFile?

File: jenkins/shared-library/vars/publish.groovy (hunk 5 above)
Comment (Wayonb) [reply jenkins/shared-library/vars/publish.groovy:57]: yea
```
sh (cat .npmrc)
14:55:20  + cat .npmrc
14:55:20  //registry.npmjs.org/:_authToken=${NPM_TOKEN}
```

**Pull request merged**